### PR TITLE
Add testing for ConstantPool class load behaviour

### PR DESCRIPTION
**Pull request by EricYangIBM**
Check that class initialize is done at the correct time when calling
ConstantPool methods.

Issue: #12157
Signed-off-by: Eric Yang <eric.yang@ibm.com>

### CODE REVIEW

**Comment by EricYangIBM** [comment]
PR tests:
https://openj9-jenkins.osuosl.org/job/Test_openjdk11_j9_sanity.functional_ppc64le_linux_Personal/132/
https://openj9-jenkins.osuosl.org/job/Test_openjdk17_j9_sanity.functional_ppc64le_linux_Personal/44/
https://openj9-jenkins.osuosl.org/job/Test_openjdk8_j9_sanity.functional_x86-64_windows_Personal/123/
https://openj9-jenkins.osuosl.org/job/Pipeline_Build_Test_JDKnext_x86-64_linux/152/

The latest force push implements the latest suggestions

**Review by babsingh — APPROVED** [review]

**Comment by tajila** [comment]
Jenkins test sanity plinux jdknext,jdk11,jdk8,jdk17

**Comment by EricYangIBM** [comment]
Failed test: https://openj9-jenkins.osuosl.org/job/Test_openjdk8_j9_sanity.functional_ppc64le_linux_Personal/86/tapResults/
Couldn't parse "1.8", changed the version 8 check to use `Double.parseDouble` instead.

File: test/functional/Java8andUp/src/org/openj9/test/constantpool/ClassInitTest.java [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+	static boolean clinitInvoked = false;
+
+	public ClassInitTest() throws Throwable {
+		String version = System.getProperty("java.vm.specification.version");
+
+		Class SharedSecrets;
+		Class JavaLangAccess;
+		Class ConstantPool;
+
+		if (Double.parseDouble(version) == 1.8) {
```
Comment (tajila) [inline test/functional/Java8andUp/src/org/openj9/test/constantpool/ClassInitTest.java:55]: @llxia Are you aware of a standard way to do these version checks in tests?

Comment (JasonFengJ9) [reply test/functional/Java8andUp/src/org/openj9/test/constantpool/ClassInitTest.java:55]: I recommend `org.openj9.test.util. VersionCheck`
https://github.com/eclipse-openj9/openj9/blob/bad687ee7ae3ff98cf547882ba4abb77e19f1b69/test/functional/TestUtilities/src/org/openj9/test/util/VersionCheck.java#L53-L58

Comment (EricYangIBM) [reply test/functional/Java8andUp/src/org/openj9/test/constantpool/ClassInitTest.java]: Changed to use VersionCheck. Tested locally on 8, 11, and 17 and passed

**Review by JasonFengJ9 — APPROVED** [review]

**Comment by tajila** [comment]
Jenkins test sanity plinux jdknext,jdk11,jdk8,jdk17

**Pull request merged**